libsrc/leddevice/dev_net/ProviderUdp.cpp

```diff
@@ -45,41 +45,42 @@ bool ProviderUdp::init(const QJsonObject &deviceConfig)
 
 		if (_address.setAddress(host) )
 		{
-			Debug( _log, "Successfully parsed %s as an ip address.", deviceConfig["host"].toString().toStdString().c_str());
+			Debug( _log, "Successfully parsed %s as an IP-address.", QSTRING_CSTR(_address.toString()));
 		}
 		else
 		{
-			Debug( _log, "Failed to parse [%s] as an ip address.", deviceConfig["host"].toString().toStdString().c_str());
-			QHostInfo info = QHostInfo::fromName(host);
-			if (info.addresses().isEmpty())
+			QHostInfo hostInfo = QHostInfo::fromName(host);
+			if ( hostInfo.error() == QHostInfo::NoError )
 			{
-				Debug( _log, "Failed to parse [%s] as a hostname.", deviceConfig["host"].toString().toStdString().c_str());
-				QString errortext = QString ("Invalid target address [%1]!").arg(host);
-				this->setInError ( errortext );
-				return false;
+				_address = hostInfo.addresses().first();
```

Comment:
Is ````addresses()```` guaranteed to be non empty ?

Reply:
Yes. The code is only executed, if no error during Hostname resolution occurred; I.e. it is existing.

QHostInfo hostInfo = QHostInfo::fromName(host);
 			if ( hostInfo.error() == QHostInfo::NoError )